### ISSUE TITLE
Bug (or confusing output): Oref0 showing 127-34 as in range when target is 106

### ISSUE THREAD

**Issue by diabeticpilot** [openaps/oref0#626]
Hello-

Today I noticed that my BG was dropping below target and saw that oref0 was not low temping or 0 temping but instead giving my regular basal. The target set in my bolus wizard on the pump is 110-110 and with autosens that was adjusted to 106. In this situation, the minPredBG was 34! and eventual COBpredBG 127 and the system was saying this was "in range" so it set my normal basal which kept pushing me lower. This continued until I put in a manual 0 temp and then suspended the pump. At each iteration, the logic was showing the same thing- that the very low minPredBG was "in range"

The reason given:

"reason":"COB: 51, Dev: 69, BGI: -17.54, ISF: 56, Target: 106, minPredBG 34, IOBpredBG 39, COBpredBG 127; 127-34 in range: no temp required; setting current basal of 1.3 as temp. "

My Setup:

Edison + Hamshield board, Jubilinix 1.1
Oref0
Had Autotune and Autosens turned on originally when I started on 8/18 but re-ran the setup script and removed Autotune from the script to turn it off, as it was putting my ISF at 35 when it is really around 55-60. 
I do not have any Oref1 features turned on
I recently ran the instructions to get pushover to work as I wanted to get alerted if carbs were required, I followed these directions: http://openaps.readthedocs.io/en/latest/docs/Customize-Iterate/oref1.html?highlight=pushover

Full Logs around the time of the decision

```
Aug 24 15:36:18 bearpanks pump-loop.log: {"carbs":134,"boluses":12.1,"mealCOB":46,"currentDeviation":13.33,"maxDeviation":21.61,"minDeviationSlope":-1.154}
Aug 24 15:36:18 bearpanks pump-loop.log: {"iob":3.087,"activity":0.0584,"bolussnooze":0,"basaliob":-0.528,"netbasalinsulin":-0.75,"hightempinsulin":0.4,"microBolusInsulin":0,"microBolusIOB":0,"time":"2017-08-24T22:35:17.000Z","lastBolusTime":1503603886000}
Aug 24 15:36:18 bearpanks pump-loop.log: {"delta":-5,"glucose":93,"short_avgdelta":-4.89,"long_avgdelta":-5.52}
Aug 24 15:36:18 bearpanks pump-loop.log: {"duration":22,"rate":1.3,"temp":"absolute"}
Aug 24 15:36:18 bearpanks pump-loop.log: Autosens adjusting basal from 1.2 to 1.3; target_bg from 110 to 106; sens from 60 to 55.6 (autosens ratio 1.08)
Aug 24 15:36:18 bearpanks pump-loop.log: Carb Impact: 11.2 mg/dL per 5m; CI Duration: 3.5 hours; remaining 4h+ CI: 0 mg/dL per 5m
Aug 24 15:36:18 bearpanks pump-loop.log: Accel. Carb Impact: 10 mg/dL per 5m; ACI Duration: 3.9 hours
Aug 24 15:36:18 bearpanks pump-loop.log: UAM Impact: 10.7 mg/dL per 5m; UAM Duration: 0.8 hours
Aug 24 15:36:18 bearpanks pump-loop.log: minPredBG: 41 minIOBPredBG: 39 minCOBPredBG: 55 avgPredBG: 41 COB: 46 carbs: 134
Aug 24 15:36:18 bearpanks pump-loop.log: BG projected to remain above 106 for 0 minutes
Aug 24 15:36:18 bearpanks pump-loop.log: bgUndershoot: -18 zeroTempDuration: 30 zeroTempEffect: 33 carbsReq: -56
Aug 24 15:36:19 bearpanks pump-loop.log: determine-basal://text/shell/enact/suggested.json
Aug 24 15:36:19 bearpanks pump-loop.log: reporting enact/suggested.json
Aug 24 15:36:19 bearpanks pump-loop.log: {"temp":"absolute","bg":93,"tick":-5,"eventualBG":124,"snoozeBG":41,"insulinReq":0,"reservoir":null,"deliverAt":"2017-08-24T22:36:18.167Z","minPredBG":999,"predBGs":{"IOB":[93,87,81,74,68,60,53,46,39,39,39,39,39],"aCOB":[93,87,81,75,70,65,61,57,54,51,49,47,46,45,44,44,44,45,47,48,51,53,56,59,63,68,72,77,82,86,91,95,99,102,105,108,111,114,116,118,120,122],"COB":[93,88,83,78,74,70,67,64,62,60,58,57,56,55,55,55,56,57,59,61,63,66,69,72,76,80,84,89,93,97,101,105,108,111,114,116,118,120,121,122,123,124]},"COB":46,"IOB":3.087,"reason":"COB: 46, Dev: 67, BGI: -16.24, ISF: 56, Target: 106, minPredBG 41, IOBpredBG 39, COBpredBG 124; 124-41 in range: no temp required, temp 1.3 ~ req 1.3U/hr. "}
Aug 24 15:36:19 bearpanks pump-loop.log: enact/enacted.json: {"insulinReq":0,"bg":107,"reservoir":null,"temp":"absolute","snoozeBG":34,"rate":1.3,"predBGs":{"COB":[107,101,95,89,84,80,75,72,68,65,63,60,59,57,56,56,56,56,57,58,60,62,64,67,70,73,77,81,86,91,95,100,104,107,111,114,117,119,122,124,126,127],"aCOB":[107,99,92,85,79,73,68,63,58,54,51,48,45,43,42,41,40,40,41,41,43,45,47,49,52,56,60,64,69,74,79,84,89,93,97,101,105,108,111,114,117,119],"IOB":[107,100,93,85,77,69,61,52,43,39,39,39,39]},"minPredBG":999,"IOB":3.661,"reason":"COB: 51, Dev: 69, BGI: -17.54, ISF: 56, Target: 106, minPredBG 34, IOBpredBG 39, COBpredBG 127; 127-34 in range: no temp required; setting current basal of 1.3 as temp. ","COB":51,"eventualBG":127,"recieved":true,"duration":30,"tick":-6,"timestamp":"2017-08-24T15:27:06.637018","deliverAt":"2017-08-24T22:26:59.060Z"}
```
Preferences.json

```
{
        "max_iob": 4,
        "max_daily_safety_multiplier": 3,
        "current_basal_safety_multiplier": 4,
        "autosens_max": 1.1,
        "autosens_min": 0.7,
        "rewind_resets_autosens": true,
        "autosens_adjust_targets": true,
        "adv_target_adjustments": true,
        "maxCOB": 120,
        "override_high_target_with_low": false,
        "skip_neutral_temps": false,
        "unsuspend_if_no_temp": false,
        "bolussnooze_dia_divisor": 3,
        "min_5m_carbimpact": 8,
        "carbratio_adjustmentratio": 1,
        "autotune_isf_adjustmentFraction": 0.5,
        "remainingCarbsFraction": 1,
        "remainingCarbsCap": 90,
        "enableUAM": false,
        "enableSMB_with_bolus": false,
        "enableSMB_with_COB": false,
        "enableSMB_with_temptarget": false
}
```

**Comment by scottleibrand** [comment]
It wasn't saying the minPredBG was "in range".  It was saying (not too clearly, apparently) that the range of minPredBG to eventualBG overlapped your target, and was therefore deciding to trust you that you knew what you were doing with your meal bolus, and that it should therefore bolus snooze and avoid counteracting it (you already had -0.5 U of basal IOB, which means the net IOB of 3 U included 3.5 U of bolus IOB).  We've had cases in the past (most recently with SMB before we introduced SMB bolus snooze) where OpenAPS zero-temps for a low minPredBG, and we invariably get complaints from people for whom that doesn't work, because it's counteracting their well-tested bolus strategy and resulting in a high BG later.  So instead, we only low temp if both your eventualBG and snoozeBG (which comes from minPredBG in this case) are below your target.

What was the eventual outcome here?  Did BG come back up above target on its own, as predicted?

It sounds like an update to oref0 might be warranted to clarify the reason field here by adding "bolus snooze" to it.  Is there anything else that you think we should be doing differently?

**Comment by diabeticpilot** [comment]
Now I understand, I would suggest adding "bolus snooze" to the reasoning so it is clear. No other change needed.

Eventual outcome was about 20 grams carb correction and I added a 0 temp basal for an hour and suspended the pump. I'm a newbie on this and my issue is trying to find out the right amount of bolus to give for protein/fat. For this bolus I took about 1/3 of the protein + fat number and put it in as carbs, but perhaps I should only count protein or figure out a different way to do it. I'll put up a post on FB or Gitter to see how low-carbers are doing that.

For this issue, should I just close it out or will you do that when you add "bolus snooze"?

THANK YOU!
